core/src/main/java/org/elasticsearch/index/query/BoostingQueryBuilder.java

```diff
@@ -172,15 +172,15 @@ public Query toQuery(QueryParseContext parseContext) throws QueryParsingExceptio
 
     @Override
```

Comment:
I see your point ok, then let's just prevent NPEs but let's not throw exception, maybe recreate the query that we got through rest in the first place, is it the one with an empty query? or without the query object completely?